docs/gitbook/development-guide/ChangeDependencies.md

```diff
@@ -0,0 +1,66 @@
+# Add or Update Dependencies 
+
+To know about the required dependencies, their minimum required versions check the requirements*.txt files.
+Whenever we pin to a specific version in requirements*.in, we add a comment explaining why we are doing so. 
+We also have comments on their use.
+
+## To install or update all dependencies
+
+**Note** : The pip-compile command lets you compile a requirements*.txt from your dependencies, specified in either setup.py or requirements.in.
+So, ensure you don’t have requirements*.txt if you compile setup.py or requirements*.in from scratch, otherwise, it might interfere.
+You can also specify a file name other than requirements.txt in the following command. 
+
+To compile all dependencies run
+```text
+pip-compile --output-file requirements.txt requirements.in
+```
+
+To update all dependencies, periodically re-run
+```text
+pip-compile --upgrade
```

Comment:
Let's change this to `make up-reqs`, which does a few more steps to make sure everything gets upgraded.

Reply:
Hi @patricksanders. Thank you for your response. Can you tell me if there is a way to run make up-reqs or I just run all the commands under it.

Reply:
It requires [GNU Make](https://www.gnu.org/software/make/), which is available on Linux and macOS (or WSL in Windows). Once you have that installed, all you have to do is run the `make up-reqs` command.

Reply:
ok on it